src/System.Net.NetworkInformation/src/System/Net/NetworkInformation/NetworkAddressChange.Linux.cs

```diff
@@ -35,21 +49,64 @@ public class NetworkChange
             {
                 lock (s_lockObj)
                 {
-                    if (s_addressChangedSubscribers == null)
+                    if (s_addressChangedSubscribers == null && s_availabilityChangedSubscribers == null)
                     {
-                        Debug.Assert(s_socket == 0, "s_socket != 0, but there are no subscribers to NetworkAddressChanged.");
+                        Debug.Assert(s_socket == 0, "s_socket != 0, but there are no subscribers to NetworkAddressChanged or NetworkAvailabilityChanged.");
                         return;
                     }
 
                     s_addressChangedSubscribers -= value;
-                    if (s_addressChangedSubscribers == null)
+                    if (s_addressChangedSubscribers == null && s_availabilityChangedSubscribers == null)
                     {
                         CloseSocket();
                     }
                 }
             }
         }
 
+        public static event NetworkAvailabilityChangedEventHandler NetworkAvailabilityChanged
+        {
+            add
+            {
+                lock (s_lockObj)
+                {
+                    if (s_socket == 0)
+                    {
+                        CreateSocket();
+                    }
+                    if (s_availabilityTimer == null)
+                    {
+                        s_availabilityTimer = new Timer(OnAvailabilityTimerFired, null, -1, -1);
```

Comment:
Nit: This will end up allocating a delegate for OnAvaialbilityTimerFired, as the language doesn't currently cache a delegate for method group conversions.  Consider caching this manually into a static.